static/dist/miyoo/app/.tmp_update/install.sh

```diff
@@ -588,73 +588,74 @@ unzip_progress() {
+    echo "Starting..." > /tmp/.update_msg
+    sleep 3
 
-echo "Starting..." > /tmp/.update_msg
-sleep 3
-
-if [ -f "/mnt/SDCARD/.tmp_update/onion.pak" ]; then
-	echo "onion.pak exists"
-	sleep 1
-else
-	echo "onion.pak is missing, extraction will fail"
-fi
+    if [ -f "/mnt/SDCARD/.tmp_update/onion.pak" ]; then
+        echo "onion.pak exists"
+        sleep 1
+    else
+        echo "onion.pak is missing, extraction will fail"
+    fi
 
-verify_file
+    verify_file
 
-sleep 1
+    sleep 1
 
-# Run the 7z extraction command in the background and redirect output to /tmp/.extraction_output
-(7z x -aoa -o"$dest" "$zipfile" -bsp1 -bb > /tmp/.extraction_output ; echo $? > "/tmp/extraction_status" ) &
+    # Run the 7z extraction command in the background and redirect output to /tmp/.extraction_output
+    (7z x -aoa -o"$dest" "$zipfile" -bsp1 -bb > /tmp/.extraction_output ; echo $? > "/tmp/extraction_status" ) &
 
-sleep 1 
+    sleep 1 
 
-if pgrep 7z > /dev/null; then
-    echo "7Z is running"
-	sleep 1
-else
-    echo "7Z is NOT running and should be"
-	sleep 1
-fi
-
-if [ -f "/mnt/SDCARD/.tmp_update/onion.pak" ]; then
-	echo "onion.pak still exists"
-	sleep 1
-else
-	echo "onion.pak is still missing, extraction has probably failed"
-	sleep 1
-	echo "the build will say it's complete but isn't"
-fi
+    if pgrep 7z > /dev/null; then
+        echo "7Z is running"
+        sleep 1
+    else
+        echo "7Z is NOT running and should be"
+        sleep 1
+    fi
 
-# Continuously update /tmp/.update_msg every 500 milliseconds until the command line finishes
-a=$(pgrep 7z)
-while [ -n "$a" ]; do
-    last_line=$(tail -n 1 /tmp/.extraction_output)
-    value=$(echo "$last_line" | sed 's/.* \([0-9]\+\)%.*/\1/')
-    if [ "$value" -eq "$value" ] 2>/dev/null; then    # check if the value is numeric
-        echo "$msg $value%" > /tmp/.update_msg
+    if [ -f "/mnt/SDCARD/.tmp_update/onion.pak" ]; then
+        echo "onion.pak still exists"
+        sleep 1
+    else
+        echo "onion.pak is still missing, extraction has probably failed"
+        sleep 1
+        echo "the build will say it's complete but isn't"
     fi
-    > /tmp/.extraction_output  # to avoid to parse a too big file
-    sleep 0.5
-    a=$(pgrep 7z)
-done
 
-# Check the exit status of the extraction command
-extraction_status=$(cat "/tmp/extraction_status")
-if [ "$extraction_status" -ne 0 ]; then
-    touch $sysdir/.installFailed
-    echo ":: Installation failed!"
-    sync
-    reboot
-    sleep 10
-    exit 0
-else
-    echo "$msg 100%" >> /tmp/.update_msg
-fi
+    # Continuously update /tmp/.update_msg every 500 milliseconds until the command line finishes
+    a=$(pgrep 7z)
+    while [ -n "$a" ]; do
+        last_line=$(tail -n 1 /tmp/.extraction_output)
+        value=$(echo "$last_line" | sed 's/.* \([0-9]\+\)%.*/\1/')
+        if [ "$value" -eq "$value" ] && [ $value -eq 0 ] 2>/dev/null; then # check if the value is numeric. If it is 0, show a different message
```

Comment:
You should nest these conditions - no need to check numeric twice. I'm also not sure the /dev/null redirect applies to the first condition.

```sh
if [ "$value" -eq "$value" ] 2>/dev/null; then # check if the value is numeric. If it is 0, show a different message
    if [ $value -eq 0 ]; then
        echo "Preparing folders..." > /tmp/.update_msg
    else
        echo "$msg $value%" > /tmp/.update_msg
    fi
fi
```

Reply:
I was thinking the same thing. couldn't come to a conclusion what was better, i'll change it!